src/arena.rs

```diff
@@ -77,7 +77,7 @@ impl<T> Handle<T> {
 /// Adding new items to the arena produces a strongly-typed [`Handle`].
 /// The arena can be indexed using the given handle to obtain
 /// a reference to the stored item.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
why do we need to clone the arena?

Reply:
I added clone because the glsl backend copies the type out of the module to later mutably borrow it and use it in the Typifier

Reply:
Where are they borrowed mutably? The backends aren't supposed to mutate the source module.

Reply:
the backend doesnt mutate but the Typifier requires a mutable reference in resolve